src/v/config/configuration.cc

```diff
@@ -1059,6 +1059,19 @@ configuration::configuration()
        .visibility = visibility::tunable},
       128_MiB,
       {.min = 16_MiB, .max = 100_GiB})
+  , storage_compaction_key_map_memory_limit_percent(
+      *this,
+      "storage_compaction_key_map_memory_limit_percent",
+      "Limit on `storage_compaction_key_map_memory`, expressed as a percentage "
+      "of memory per shard, that bounds the amount of memory used by "
+      "compaction key-offset maps. NOTE: Memory per shard is computed after "
+      "`wasm_per_core_memory_reservation`. Only respected when "
```

Comment:
A note, maybe for the future

> NOTE: Memory per shard is computed after `wasm_per_core_memory_reservation

I'm guessing this sort of annotation isn't going to be scalable

Reply:
Yeah fair, though opted to include for clarity since it may be surprising otherwise (though tbh, my hope is that few people will touch this config)